bnr-utils/nz_azConnector/nz_azConnector.go

```diff
@@ -1,500 +1,506 @@
+
+				// Set up file to download the blob to
+				dir, filename := filepath.Split(blobInfo.Name)
+
+				relfilepath, err := filepath.Rel(uniqueid, dir)
+				if err != nil {
+					return fmt.Errorf("Error in fetching download relative path: %v", err)
+				}
+
+				dumpdir := filepath.Join(outdir, relfilepath)
+				err = os.MkdirAll(dumpdir, 0777)
+				if err != nil {
+					return fmt.Errorf("Error in creating backup directory structure: %v", err)
+				}
+
+				switch filename {
+				case "locations.txt":
+					locations = append(locations, path.Join(dumpdir, filename))
+				case "contents.txt":
+					contents = append(contents, path.Join(dumpdir, filename))
+				}
+
+				outfilepath := path.Join(dumpdir, filename)
+				j := downloadJob{conn: *cn, outfilepath: outfilepath, blobname: blobInfo.Name}
+				work <- &j
+
+				blobfound--
+			}
+		}
+
+		if blobfound == 0 {
+			return fmt.Errorf("No matching blob found. Please check if DB name, hostname, uniqueid or containername are correct. If error persists contact IBM support team. Azaccount:%s AzContainer:%s Blobpath:%s, Uniqueid:%s", cn.azaccount, cn.azcontainer, blobpath, uniqueid)
+		}
+	}
+	close(work)
+	<-done
+	log.Println("Total files downloaded:", filesdownloaded)
+	updateLocation(locations, outdir)
+	updateContents(contents)
+	return err
+}
+
+func updateLocation(arrLoc []string, outdir string) {
+	for _, locFile := range arrLoc {
+		input, err := ioutil.ReadFile(locFile)
+		if err != nil {
+			log.Fatalf("Unable to open %s to read: %v\n", locFile, err)
+		}
+		lines := strings.Split(string(input), "\n")
+		if len(lines) == 2 && !strings.HasSuffix(lines[len(lines)-2], outdir) {
+			f, err := os.OpenFile(locFile,
+				os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+			if err != nil {
+				log.Fatalf("Unable to open %s for update: %v\n", locFile, err)
+			}
+			defer f.Close()
+			textAppend := "1,1,1," + outdir + "\n"
+			if _, err := f.WriteString(textAppend); err != nil {
+				log.Fatalf("Unable to update %s: %v\n", locFile, err)
+			}
+		}
+	}
+}
+
+func updateContents(arrContents []string) {
+	for _, contentFile := range arrContents {
+		input, err := ioutil.ReadFile(contentFile)
+		if err != nil {
+			log.Fatalln("Unable to open %s to read: %v\n", contentFile, err)
+		}
+
+		lines := strings.Split(string(input), "\n")
+		var textline []string
+		for i := 0; i < len(lines); i++ {
+			line := lines[i]
+			token := strings.Split(line, ",")
+			if token[len(token)-1] == "0" {
+				token[len(token)-1] = "1"
+			}
+			textline = append(textline, strings.Join(token, ","))
+		}
+		output := strings.Join(textline, "\n")
+		err = ioutil.WriteFile(contentFile, []byte(output), 0644)
+		if err != nil {
+			log.Fatalln("Unable to update %s: %v\n", contentFile, err)
+		}
+	}
+}
+
+func main() {
+	var conn Conn
+	var backupinfo BackupInfo
+	var othargs OtherArgs
+
+	// parse input args
+	parseArgs(&conn, &backupinfo, &othargs)
+	flag.Parse()
+
+	//Checking whether '-' is passed or not in flags
```

Comment:
Both these checks should be after MultiWriter initialisation. Errors should not be printed only to stdout, but also to log file.

Reply:
Done with these changes.